android/sample/src/main/kotlin/com/gu/source/IconsPreview.kt

```diff
@@ -0,0 +1,115 @@
+import com.gu.source.presets.palette.Neutral0
+import com.gu.source.presets.palette.Neutral100
+import com.gu.source.presets.palette.Sport500
+import com.gu.source.presets.typography.HeadlineBold20
+import com.gu.source.presets.typography.TextSansBold14
+import com.gu.source.presets.typography.TextSansBold15
+
+private val icons = mapOf(
+    "Base" to listOf(
+        Source.Icons.Base.Check,
+        Source.Icons.Base.ChevronLeft,
+        Source.Icons.Base.ChevronRight,
+        Source.Icons.Base.Minus,
+        Source.Icons.Base.Plus,
+    ),
+)
+
+@Composable
+internal fun Icons(modifier: Modifier = Modifier) {
+    Surface(
+        modifier = modifier,
+        color = AppColour(
+            Source.Palette.Neutral100,
+            Source.Palette.Neutral0,
+        ).current,
+        contentColor = AppColour(
+            Source.Palette.Neutral0,
+            Source.Palette.Neutral100,
+        ).current,
+    ) {
+        LazyVerticalGrid(
+            columns = GridCells.Fixed(2),
+            contentPadding = PaddingValues(vertical = 8.dp),
+        ) {
+            item {
+                Text(
+                    text = "Icons",
+                    style = Source.Typography.HeadlineBold20,
+                    modifier = Modifier.padding(8.dp),
+                )
+            }
+            icons.keys.forEachIndexed { iconIndex, iconKey ->
+                item(span = { GridItemSpan(2) }) {
+                    Column {
+                        Spacer(modifier = Modifier.height(8.dp))
+                        HorizontalDivider()
+                        Text(
+                            text = iconKey,
+                            style = Source.Typography.TextSansBold14,
+                            modifier = Modifier.padding(8.dp),
+                        )
+                    }
+                }
+                items(icons[iconKey].orEmpty()) { icon ->
+                    Box(
+                        modifier = Modifier
+                            .height(50.dp)
+                            .fillMaxWidth(),
+                    ) {
+                        Text(
+                            text = icon.name,
+                            modifier = Modifier
+                                .align(Alignment.Center),
+                            style = Source.Typography.TextSansBold15,
+                        )
+                        SourceIconButton(
+                            icon = icon,
+                            iconColor = AppColour(
+                                Source.Palette.Brand400,
+                                Source.Palette.Sport500,
+                            ),
+                            priority = SourceButton.Priority.TertiaryOnWhite,
+                            contentDescription = null,
+                            onClick = {},
+                            size = SourceButton.Size.XSmall,
+                        )
+                    }
+                }
```

Comment:
This gets simplified a lot once you remove the "Base" label and convert map to key

```suggestion
            item(span = { GridItemSpan(2) }) {
                Text(
                    text = "Icons",
                    style = Source.Typography.HeadlineBold20,
                    modifier = Modifier.padding(8.dp),
                )
            }
            items(icons) { icon ->
                Row {
                    SourceIconButton(
                        icon = icon,
                        iconColor = AppColour(
                            Source.Palette.Brand400,
                            Source.Palette.Sport500,
                        ),
                        priority = SourceButton.Priority.TertiaryOnWhite,
                        contentDescription = null,
                        onClick = {},
                        size = SourceButton.Size.XSmall,
                    )
                    Text(
                        text = icon.name,
                        modifier = Modifier.align(Alignment.CenterVertically),
                        style = Source.Typography.TextSansBold15,
                    )
                }
            }
        }
```